File: web/src/api/qrcode.js
import service from '@/utils/request'

export const getQrCodeList = (params) => {
    return service({
        url: '/qrcode/list',
        method: 'get',
        params
    })
}

export const deleteQrCode = (params) => {
    return service({
        url: '/qrcode/delete',
        method: 'delete',
        params,
    })
}

export const updateQrCode = (data) => {
    return service({
        url: '/qrcode/update',
        method: 'put',
        data
    })
}

export const createQrCode = (data) => {
    return service({
        url: '/qrcode/create',
        method: 'post',
        data
    })
}

export const downloadQrCode = (params) => {
    return service({
        url: '/qrcode/download',
        method: 'get',
        params
    })
}

File: server/router/business/qrcode.go
package business

import (
	v1 "cooller/server/api/v1"
	"github.com/gin-gonic/gin"
)

type QrCodeRouter struct{}

func (e *QrCodeRouter) InitQrCodeRouter(Router *gin.RouterGroup, RouterPub *gin.RouterGroup) {
	qrcodePublicRouterWithoutRecord := RouterPub.Group("qrcode")

	qrcodeApi := v1.ApiGroupApp.BusinessApiGroup.QrCodeApi
	{
		qrcodePublicRouterWithoutRecord.GET("list", qrcodeApi.GetQrCodeList)
		qrcodePublicRouterWithoutRecord.POST("create", qrcodeApi.CreateQrCode)
		qrcodePublicRouterWithoutRecord.PUT("update", qrcodeApi.UpdateQrCode)
		qrcodePublicRouterWithoutRecord.DELETE("delete", qrcodeApi.DeleteQrCodeById)
		qrcodePublicRouterWithoutRecord.GET("download", qrcodeApi.DownloadQrCodeFile)
		qrcodePublicRouterWithoutRecord.GET("scan", qrcodeApi.ScanQrCodeFile)
	}
}
